### PR TITLE
add validation plugin and tests

### PR DESCRIPTION
**Pull request by monya007**

### CODE REVIEW

File: src/validation.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		}
+
+		const min = parseInt( result.info.configuration.min );
+
+		this.tooltipView.isVisible = true;
+
+		const viewElement = this.editor.editing.mapper.toViewElement( result.element );
+		const domElement = this.editor.editing.view.domConverter.mapViewToDom( viewElement );
+
+		const match = domElement.innerHTML.match( validation );
```
Comment (pmelab) [inline src/validation.js]: The elements innerHTML has to be trimmed before checking. Sometimes whitespaces add up there and we end up with more characters that are actually there.

File: tests/manual/validation/validation.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+ClassicEditor
+	.create( document.querySelector( '#editor' ), {
+		plugins: [ EnterPlugin, TypingPlugin, ParagraphPlugin, HeadingPlugin, TemplatePlugin, UndoPlugin, Validation ],
+		toolbar: [ 'heading', '|', 'template', '|', 'undo', 'redo' ],
+		templates: {
+			text: {
+				label: 'text',
+				template: '<p class="simple" ck-type="text" ck-min="2">Limited</p>',
+				validation: /./gm
```
Comment (pmelab) [inline tests/manual/validation/validation.js]: I suggest moving this into a `ck-validate` attribute. This should be added to an elements `configuration` property automatically. And then multiple text fields within one element can have different validation expressions.

File: tests/manual/validation/validation.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+			},
+			textrequired: {
+				label: 'text required',
+				template: '<p class="simple" ck-type="text">Limited</p>',
+			},
+			container: {
+				label: 'Container',
+				template: `<div class="wrapper">
+							<div class="container" ck-type="container" ck-contains="text textrequired" ck-min="1"></div>
```
Comment (pmelab) [inline tests/manual/validation/validation.js]: We don't need to handle containers in this PR. There already is another issue open for container min/max handling.

File: src/validation.js (hunk 1 above)
Comment (monya007) [reply src/validation.js]: Updated

File: tests/manual/validation/validation.js (hunk 2 above)
Comment (monya007) [reply tests/manual/validation/validation.js]: Moved to `ck-validate` attribute, also keep `ck-min` attribute

File: tests/manual/validation/validation.js (hunk 3 above)
Comment (monya007) [reply tests/manual/validation/validation.js]: Removed containers from tests

**Pull request merged**